scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamFunctions.scala

```diff
@@ -22,4 +25,40 @@ object BeamFunctions {
     def processElement(c: DoFn[A, B]#ProcessContext): Unit =
       c.output(f(c.element()))
   }
+
+  case class MapSideAggregator[K, V](size: Int, semigroup: Semigroup[V]) extends DoFn[KV[K, V], KV[K, V]] {
```

Comment:
Do we really want to keep it over `KV`, and not just `()`?

Reply:
I agree that if beam doesn't require it, you are probably better staying with `Tuple2` and making a concrete `Tuple2Coder` so you don't have to allocate wasted tuple or KV instances.